scene/gui/code_edit.cpp

```diff
@@ -2004,7 +2004,7 @@ void CodeEdit::confirm_code_completion(bool p_replace) {
 		return;
 	}
 
-	char32_t caret_last_completion_char;
+	char32_t caret_last_completion_char = 0;
```

Comment:
Not sure why this warning wasn't triggered before, but for some reason I had to fix this in order to pass CI.

<details>
<summary>CI log</summary>

```
In file included from ./core/templates/pair.h:34,
from ./core/templates/rb_map.h:36,
from ./scene/main/node.h:35,
from ./scene/main/canvas_item.h:34,
from ./scene/gui/control.h:37,
from ./scene/gui/text_edit.h:34,
from scene/gui/code_edit.h:34,
from scene/gui/code_edit.cpp:31:
./core/templates/hashfuncs.h: In member function 'void CodeEdit::confirm_code_completion(bool)':
./core/templates/hashfuncs.h:176:4: error: 'caret_last_completion_char' may be used uninitialized in this function [-Werror=maybe-uninitialized]
176 | h ^= h >> 16;
| ~~^~~~~~~~~~
scene/gui/code_edit.cpp:2007:11: note: 'caret_last_completion_char' was declared here
2007 | char32_t caret_last_completion_char;
| ^~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
scons: *** [scene/gui/code_edit.linuxbsd.editor.x86_64.o] Error 1
scons: building terminated because of errors.
```
</details>